main/externalData/inventory/tokens.ts

```diff
@@ -131,7 +99,11 @@ export default class TokenLoader {
     }
   }
 
-  getTokens (chainId: number) {
-    return this.tokenList.filter(token => token.chainId === chainId)
+  getTokens (chains: number[]) {
+    return this.tokens.filter(token => !token.extensions?.omit && chains.includes(token.chainId))
+  }
+
+  getBlacklist (chains: number[]) {
```

Comment:
can make chains default to `[]` here so we don't need to pass in an empty array when we want the entire list